packages/emitter-framework/src/python/components/class-declaration/class-declaration.test.tsx

```diff
@@ -0,0 +1,1334 @@
+      }
+  
+      model ${t.model("Widget")} {
+        id: string;
+        primaryColor: Color.red | Color.blue;
+      }
+      `);
+
+    expect(
+      getOutput(program, [<EnumDeclaration type={Color} />, <ClassDeclaration type={Widget} />]),
+    ).toRenderTo(
+      `
+      from dataclasses import dataclass
+      from enum import StrEnum
+      from typing import Literal
+
+      class Color(StrEnum):
+        RED = "RED"
+        BLUE = "BLUE"
+        GREEN = "GREEN"
+
+
+      @dataclass(kw_only=True)
+      class Widget:
+        id: str
+        primary_color: Literal[Color.RED, Color.BLUE]
+
+      `,
+    );
+  });
+
+  it("handles a union of integer literals in a class property", async () => {
+    const { program, Widget } = await Tester.compile(t.code`
+      model ${t.model("Widget")} {
+        id: string;
+        priority: 1 | 2 | 3;
+      }
+      `);
+
+    expect(getOutput(program, [<ClassDeclaration type={Widget} />])).toRenderTo(
+      `
+      from dataclasses import dataclass
+      from typing import Literal
+
+      @dataclass(kw_only=True)
+      class Widget:
+        id: str
+        priority: Literal[1, 2, 3]
+
+      `,
+    );
+  });
+
+  it("handles a union of boolean literals in a class property", async () => {
+    const { program, Widget } = await Tester.compile(t.code`
+      model ${t.model("Widget")} {
+        id: string;
+        isActiveOrEnabled: true | false;
+      }
+      `);
+
+    expect(getOutput(program, [<ClassDeclaration type={Widget} />])).toRenderTo(
+      `
+      from dataclasses import dataclass
+      from typing import Literal
+
+      @dataclass(kw_only=True)
+      class Widget:
+        id: str
+        is_active_or_enabled: Literal[True, False]
+
+      `,
+    );
+  });
+
+  it("handles a mixed union of literals and variant references", async () => {
+    const { program, Color, Widget } = await Tester.compile(t.code`
+      union ${t.union("Color")} {
+        red: "RED",
+        blue: "BLUE",
+      }
+  
+      model ${t.model("Widget")} {
+        id: string;
+        mixedValue: "custom" | 42 | true | Color.red;
+      }
+      `);
+
+    expect(
+      getOutput(program, [<EnumDeclaration type={Color} />, <ClassDeclaration type={Widget} />]),
+    ).toRenderTo(
+      `
+      from dataclasses import dataclass
+      from enum import StrEnum
+      from typing import Literal
+
+      class Color(StrEnum):
+        RED = "RED"
+        BLUE = "BLUE"
+
+
+      @dataclass(kw_only=True)
+      class Widget:
+        id: str
+        mixed_value: Literal["custom", 42, True, Color.RED]
+
+      `,
+    );
+  });
+
+  it("renders a never-typed member as typing.Never", async () => {
+    const { program, Widget } = await Tester.compile(t.code`
+    model ${t.model("Widget")} {
+      property: never;
+    }
+    `);
+
+    expect(getOutput(program, [<ClassDeclaration type={Widget} />])).toRenderTo(`
+      from dataclasses import dataclass
+      from typing import Never
```

Comment:
`typing.Never` was only added in 3.11, so depending on the proposed minimum supported version (for Azure SDK that's 3.9), this might need to be imported from typing-extensions or use the equivalent `typing.NoReturn`